sequencer/src/test/java/net/consensys/linea/sequencer/txselection/selectors/MaxBundleBlockGasTransactionSelectorTest.java

```diff
@@ -0,0 +1,180 @@
+
+    final var mockTxs = params.stream().map(__ -> mockTransaction()).toList();
+    final var bundle =
+        new BundlePoolService.TransactionBundle(
+            Hash.wrap(Bytes32.repeat((byte) seq++)),
+            mockTxs,
+            1L,
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty());
+    final var evaluationContexts =
+        bundle.pendingTransactions().stream()
+            .map(pt -> new TestTransactionEvaluationContext(mock(ProcessableBlockHeader.class), pt))
+            .toList();
+
+    for (int i = 0; i < params.size(); i++) {
+      final var mockTransactionProcessingResult =
+          mockTransactionProcessingResult(params.get(i).gasUsedByTransaction);
+      verifyTransactionSelection(
+          transactionSelector,
+          evaluationContexts.get(i),
+          mockTransactionProcessingResult,
+          params.get(i).selectionResult);
+      assertThat(
+              (BundleGasTracker) selectorsStateManager.getSelectorWorkingState(transactionSelector))
+          .isEqualTo(
+              new BundleGasTracker(
+                  params.get(i).expectedBlockBundleGasUsed, params.get(i).expectedBundleGasUsed));
+    }
+  }
+
+  private void verifyTransactionSelection(
+      final PluginTransactionSelector selector,
+      final TestTransactionEvaluationContext evaluationContext,
+      final TransactionProcessingResult processingResult,
+      final TransactionSelectionResult expectedSelectionResult) {
+    var selectionResult =
+        selector.evaluateTransactionPostProcessing(evaluationContext, processingResult);
+    assertThat(selectionResult).isEqualTo(expectedSelectionResult);
+  }
+
+  private Transaction mockTransaction() {
+    Transaction transaction = mock(Transaction.class);
+    when(transaction.getHash()).thenReturn(Hash.wrap(Bytes32.repeat((byte) seq++)));
+    return transaction;
+  }
+
+  private TransactionProcessingResult mockTransactionProcessingResult(long gasUsedByTransaction) {
+    TransactionProcessingResult mockTransactionProcessingResult =
+        mock(TransactionProcessingResult.class);
+    when(mockTransactionProcessingResult.getEstimateGasUsedByTransaction())
+        .thenReturn(gasUsedByTransaction);
+    return mockTransactionProcessingResult;
+  }
+
+  record TestParams(
+      long gasUsedByTransaction,
+      TransactionSelectionResult selectionResult,
+      long expectedBlockBundleGasUsed,
+      long expectedBundleGasUsed) {
+
+    static List<TestParams> selected(final long... gasUsedByTransactions) {
+      final List<TestParams> params = new ArrayList<>(gasUsedByTransactions.length);
+
+      long cumulativeBundleGasUsed = 0;
+
+      for (final long gasUsedByTransaction : gasUsedByTransactions) {
+        cumulativeBundleGasUsed += gasUsedByTransaction;
+        params.add(
+            new TestParams(
+                gasUsedByTransaction, SELECTED, cumulativeBundleGasUsed, cumulativeBundleGasUsed));
```

Comment:
these two gasUsed values are always the same - this is because we're only testing with one bundle, right? If we had two bundles they could be different

Reply:
correct